### PR TITLE
SA-1715 - (UI) Comparators - Update visible active filters for new comparators

### PR DESCRIPTION
**Pull request by nicklemmon**
[SA-1715](https://sparkpost.atlassian.net/browse/SA-1715)

### What Changed
- Updates review badger
- Adds new global `Comparison` text component - updates one-off component in `FiltersForm` with said component
- Adds new `ActiveFiltersV2` component to `ReportOptions`
- Adds new `removeFilterV2` that uses `reducerV2` in the report builder context hook
- Migrates report builder helper functions to a single file, adds JSDoc comments to clarify their purpose. I renamed a few of the helpers as well for clarity's sake.
- I tweaked a few Cypress tests while I was there to make them a little better - also added some missing `id` attributes to form elements in the saved reports modal. That was an a11y bug.

### How To Test
1. Verify filter tags render when filters are applied via the URL - append `query_filters=%255B%257B%2522AND%2522%3A%257B%2522campaigns%2522%3A%257B%2522like%2522%3A%255B%2522hello%2522%2C%2522world%2522%255D%257D%2C%2522templates%2522%3A%257B%2522eq%2522%3A%255B%2522greg-hackathon%2522%255D%2C%2522notEq%2522%3A%255B%2522gregs-test%2522%255D%257D%257D%257D%2C%257B%2522AND%2522%3A%257B%2522sending_ips%2522%3A%257B%2522like%2522%3A%255B%2522hello%2522%255D%2C%2522notLike%2522%3A%255B%2522hello-again%2522%255D%257D%257D%257D%255D` to the `/signals/analytics` URL
1. Use the `FiltersForm` to add and remove filters. Verify the tags update accordingly
1. Use the remove buttons within the tags and verify values are accurately removed
1. Verify the saved report modal renders the tags appropriately

### To Do
- [x] Add supporting tests
- [ ] Incorporate feedback

### CODE REVIEW

File: cypress/tests/integration/signals-analytics/analytics-report/constants/index.js [hunk 1]
```diff
@@ -142,10 +142,18 @@ export const METRICS = [
...
+    queryParam: 'count_initial_rendered',
+  },
   {
     name: 'Open Rate',
     queryParam: 'open_rate_approx',
   },
+  {
+    name: 'Opens',
+    queryParam: 'count_rendered',
+  },
```
Comment (nicklemmon) [inline cypress/tests/integration/signals-analytics/analytics-report/constants/index.js:156]: These checkboxes were being skipped by Cypress - looks like they were added after this spec was first introduced

File: src/pages/reportBuilder/components/ReportOptions.js [hunk 2]
```diff
@@ -18,40 +18,14 @@ import DateTimeSection from './DateTimeSection';
 import useRouter from 'src/hooks/useRouter';
 import { selectCondition } from 'src/selectors/accessConditionState';
 import { isUserUiOptionSet } from 'src/helpers/conditions/user';
-import { dehydrateFilters } from '../helpers';
-
-export const ActiveFilters = ({ filters, handleFilterRemove }) => {
```
Comment (nicklemmon) [inline src/pages/reportBuilder/components/ReportOptions.js:23]: Moved this to the bottom of the file, and hoisted it using a `function` declaration. Should we move these to separate files at this point?

File: src/pages/reportBuilder/context/ReportBuilderContext.js [hunk 3]
```diff
@@ -100,7 +100,7 @@ const reducer = (state, action) => {
     }
 
     default:
-      return state;
+      throw new Error(`${action.type} is not supported.`);
```
Comment (nicklemmon) [inline src/pages/reportBuilder/context/ReportBuilderContext.js:103]: I feel like it's better to go ahead and introduce this? Should help us out as the feature grows.

File: src/pages/reportBuilder/helpers/tests/helpers.test.js [hunk 4]
```diff
@@ -44,9 +45,39 @@ const GROUPINGS = [
...
+            subaccounts: {
+              notEq: [{ value: 'should be gone', id: 123, name: 'Should not be here' }],
+            },
+          },
+        },
+      ]);
+      expect(dehydratedFilters).toMatchSnapshot();
+    });
+  });
+
```
Comment (nicklemmon) [inline src/pages/reportBuilder/helpers/tests/helpers.test.js:77]: These were migrated from the other file.

File: src/pages/reportBuilder/helpers/helpers.test.js [hunk 5]
```diff
@@ -1,31 +0,0 @@
-import * as helpers from '.';
```
Comment (PatrickSison) [inline src/pages/reportBuilder/helpers/helpers.test.js:1]: Did we just remove these tests because they're covered by the integration testing?

Comment (PatrickSison) [reply src/pages/reportBuilder/helpers/helpers.test.js:1]: Nevermind. Saw it moved

**Review by PatrickSison — APPROVED** [review]
Everything seems to look good

**Pull request merged**